js/Track/Model/File/BED.js

```diff
@@ -19,15 +35,27 @@ Genoverse.Track.Model.File.BED = Genoverse.Track.Model.File.extend({
           color = this.scoreColor(isNaN(score) ? 1000 : score);
         }
 
-        this.insertFeature({
-          chr             : chr,
-          start           : parseInt(fields[1], 10),
-          end             : parseInt(fields[2], 10),
-          id              : chr + ':' + fields[1] + '-' + fields[3],
-          label           : fields[3],
-          color           : color,
-          originalFeature : fields
-        });
+        if(len == 12){ //subfeatures present
+          feature.blockCount = parseInt(fields[9],10);
+          var subfeatures = [];
+          var blockSizes = fields[10].split(",");
+          var blockStarts = fields[11].split(",");
+          blockSizes.pop();
+          blockStarts.pop();
```

Comment:
because the blockSizes and blockStarts fields are of the form 1,2,3,4, 
the ending comma leads to an extra empty element when we split the above string using ',' as delimiter.